solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/HeuristicBoostedModel.java

```diff
@@ -49,18 +49,18 @@
     public float weight;
     public BoostType type;
 
-    public EuristicFeatureBoost(Map<String, Object> featureBoostParam) {
+    public HeuristicFeatureBoost(Map<String, Object> featureBoostParam) {
       checkParams(featureBoostParam);
       feature = ((String)featureBoostParam.get(FEATURE_NAME));
       weight = NamedParams.convertToFloat(featureBoostParam.get(WEIGHT));
       featureIndex = getFeatureIndex(feature);
-      type = BoostType.product;
+      type = BoostType.PRODUCT;
 
       if (featureBoostParam.get(BOOST_TYPE) != null) {
         String fetchedType = (String) featureBoostParam.get(BOOST_TYPE);
-        if (BoostType.sum.toString().equals(fetchedType)) {
-          type = BoostType.sum;
-        } else if ((!BoostType.sum.product.toString().equals(fetchedType))) {
+        try {
+          type = BoostType.valueOf(fetchedType);
```

Comment:
minor: I would add a to `fetchedType.strip().toUpperCase()` so that is case insensitive in the conf file..